### PR TITLE
WIP #598: Fixed packaging of hugo template

### PR DESCRIPTION
**Pull request by otbe**

### CODE REVIEW

File: factcast-schema-registry-cli/src/main/kotlin/org/factcast/schema/registry/cli/fs/FileSystemServiceImpl.kt [hunk 1]
```diff
@@ -66,10 +69,38 @@ class FileSystemServiceImpl : FileSystemService {
         return Files.readAllBytes(file)
     }
 
+    override fun copyFromClasspath(source: String, target: Path) {
+        val url = javaClass.classLoader.getResource(source) ?: throw Exception("didnt found '$source' on classpath")
```
Comment (uweschaefer) [inline factcast-schema-registry-cli/src/main/kotlin/org/factcast/schema/registry/cli/fs/FileSystemServiceImpl.kt]: Codacy Issue found: [TooGenericExceptionThrown - [url] at /src/factcast-schema-registry-cli/src/main/kotlin/org/factcast/schema/registry/cli/fs/FileSystemServiceImpl.kt:73:64](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5032643)

**Comment by uweschaefer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5032643&bid=16580386)

File: factcast-schema-registry-cli/src/main/kotlin/org/factcast/schema/registry/cli/fs/FileSystemServiceImpl.kt [hunk 2]
```diff
@@ -66,10 +69,38 @@ class FileSystemServiceImpl : FileSystemService {
...
+
+        return when (val urlConnection: URLConnection = url.openConnection()) {
+            is JarURLConnection -> copyJarResourcesRecursively(target.toFile(), urlConnection)
+            is FileURLConnection -> FileUtils.copyDirectory(File(url.path), target.toFile())
+            else -> throw Exception("not supported")
```
Comment (uweschaefer) [inline factcast-schema-registry-cli/src/main/kotlin/org/factcast/schema/registry/cli/fs/FileSystemServiceImpl.kt]: Codacy Issue found: [TooGenericExceptionThrown - [copyFromClasspath] at /src/factcast-schema-registry-cli/src/main/kotlin/org/factcast/schema/registry/cli/fs/FileSystemServiceImpl.kt:78:21](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5032643)

File: factcast-schema-registry-cli/src/main/kotlin/org/factcast/schema/registry/cli/fs/FileSystemServiceImpl.kt [hunk 3]
```diff
@@ -66,10 +69,38 @@ class FileSystemServiceImpl : FileSystemService {
...
+    }
+
     override fun copyDirectory(from: Path, to: Path) {
         FileUtils.copyDirectory(from.toFile(), to.toFile())
     }
 
     private fun list(path: Path) =
         Files.list(path).toList()
+
+    private fun copyJarResourcesRecursively(destination: File, jarConnection: JarURLConnection) {
```
Comment (uweschaefer) [inline factcast-schema-registry-cli/src/main/kotlin/org/factcast/schema/registry/cli/fs/FileSystemServiceImpl.kt]: Codacy Issue found: [NestedBlockDepth - 4/4 - [copyJarResourcesRecursively] at /src/factcast-schema-registry-cli/src/main/kotlin/org/factcast/schema/registry/cli/fs/FileSystemServiceImpl.kt:89:17](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5032643)

File: factcast-schema-registry-cli/src/main/kotlin/org/factcast/schema/registry/cli/fs/FileSystemServiceImpl.kt (hunk 2 above)
Comment (uweschaefer) [reply factcast-schema-registry-cli/src/main/kotlin/org/factcast/schema/registry/cli/fs/FileSystemServiceImpl.kt]: i agree. throw new IOException("Unsupported connection type: "+url.Connection.javaClass); ?

File: factcast-schema-registry-cli/src/main/kotlin/org/factcast/schema/registry/cli/fs/FileSystemServiceImpl.kt (hunk 1 above)
Comment (uweschaefer) [reply factcast-schema-registry-cli/src/main/kotlin/org/factcast/schema/registry/cli/fs/FileSystemServiceImpl.kt]: change to IOException

File: factcast-schema-registry-cli/pom.xml [hunk 4]
```diff
@@ -108,6 +108,11 @@
       <artifactId>json-schema-validator</artifactId>
       <version>2.2.12</version>
     </dependency>
+    <dependency>
```
Comment (uweschaefer) [inline factcast-schema-registry-cli/pom.xml]: necessary?

**Comment by uweschaefer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5032643&bid=16580386)

File: factcast-schema-registry-cli/src/main/kotlin/org/factcast/schema/registry/cli/fs/FileSystemServiceImpl.kt [hunk 5]
```diff
@@ -66,10 +69,39 @@ class FileSystemServiceImpl : FileSystemService {
...
```
Comment (uweschaefer) [inline factcast-schema-registry-cli/src/main/kotlin/org/factcast/schema/registry/cli/fs/FileSystemServiceImpl.kt]: Codacy Issue found: [NestedBlockDepth - 4/4 - [copyJarResourcesRecursively] at /src/factcast-schema-registry-cli/src/main/kotlin/org/factcast/schema/registry/cli/fs/FileSystemServiceImpl.kt:89:17](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5032643)

File: factcast-schema-registry-cli/pom.xml (hunk 4 above)
Comment (otbe) [reply factcast-schema-registry-cli/pom.xml]: Nope. Thx 👍

**Comment by uweschaefer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5032643&bid=16580386)

File: factcast-schema-registry-cli/src/main/kotlin/org/factcast/schema/registry/cli/fs/FileSystemServiceImpl.kt (hunk 5 above)
Comment (uweschaefer) [inline factcast-schema-registry-cli/src/main/kotlin/org/factcast/schema/registry/cli/fs/FileSystemServiceImpl.kt]: Codacy Issue found: [NestedBlockDepth - 4/4 - [copyJarResourcesRecursively] at /src/factcast-schema-registry-cli/src/main/kotlin/org/factcast/schema/registry/cli/fs/FileSystemServiceImpl.kt:89:17](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=5032643)

**Pull request merged**